delta_backend/src/ConversionChecker.py

```diff
@@ -266,3 +371,45 @@ def _convertToOnlyIfTo(self, expressionRule, fieldName, fieldValue, summarise, r
             if report_unexpected_exception:
                 message = ExceptionMessages.MESSAGES[ExceptionMessages.UNEXPECTED_EXCEPTION] % (e.__class__.__name__, e)
                 return message
+    
+    # Check if Snomed code is numeric and reject other forms
+    def _convertToSnomed(self, expressionRule, fieldName, fieldValue, summarise, report_unexpected_exception):
+        """
+        Validates that a SNOMED code is a non-empty string containing only digits.
+        """
+        try:
+            if not fieldValue:
+                return fieldValue
+            if not isinstance(fieldValue, str) or not fieldValue.isdigit():
+                raise ValueError(f"Invalid SNOMED code: {fieldValue}")
+            return fieldValue
+        except Exception as e:
+            if report_unexpected_exception:
+                message = ExceptionMessages.MESSAGES[ExceptionMessages.UNEXPECTED_EXCEPTION] % (e.__class__.__name__, e)
+                self.errorRecords.append({
+                    "field": fieldName,
+                    "value": fieldValue,
+                    "message": message
+                })
+            return ""
+
+    # Check if Input is boolean or if input is a string with true or false, convert to Boolean
+    def _convertToBoolean(self, expressionRule, fieldName, fieldValue, summarise, report_unexpected_exception):
+        try:
+            if isinstance(fieldValue, bool):
+                return fieldValue
+
+            if isinstance(fieldValue, str):
+                lowered = fieldValue.strip().lower()
+                if lowered == "true":
+                    return True
+                elif lowered == "false":
+                    return False
+            return "" 
```

Comment:
Should it not be an error if we receive something other than a `bool` or `str`, or if we receive an invalid string?